### PR TITLE
Implements `-q[l|m|h|k]`

### PR DESCRIPTION
**Pull request by jvdoorn**
## List of Changes
- Implemented `-q/--quality [k,h,m,l]` flags
- Removed `-k/-e/-m/-l` flags

## Motivation
Implements behaviour as described in #146 except for `-qp`. This flag would specify default behaviour and seems redundant to me. It could however be easily implemented if that is prefered for this PR.

## Testing Status
Old tests have been updated to use new flags when needed and a new test has been written to test the flags.

## Further Comments
The new test might be a bit too extensive but it covers every way someone could write the flags (par example `-q h` or `-qh` are both tested). The production quality does not have its own section in the config but instead reads from the `CLI` section (which was its old behaviour).

## Acknowledgement
- [x] I have read the [Contributing Guidelines](https://github.com/ManimCommunity/manim/wiki/Documentation-guidelines-(WIP))

### CODE REVIEW

File: manim/config/config.py [hunk 1]
```diff
@@ -80,13 +80,11 @@ def _parse_config(config_parser, args):
...
-            section = config_parser[flag]
-            break
-    else:
-        section = config_parser["CLI"]
-    config = {opt: section.getint(opt) for opt in config_parser[flag]}
+    quality = _determine_quality(args)
+    section = config_parser[quality if quality != "production" else "CLI"]
+
+    # Loop over low quality for the keys, could be any quality really
+    config = {opt: section.getint(opt) for opt in config_parser['low_quality']}
```
Comment (leotrs) [inline manim/config/config.py:87]: Very clean code, I like it.

File: manim/constants.py [hunk 2]
```diff
@@ -126,3 +126,11 @@ class MyText(Text):
...
     "executable with the --js_renderer_path flag."
 )
+
+# Video qualities
+QUALITIES = {
+    'fourk_quality': 'k',
+    'high_quality': 'h',
+    'medium_quality': 'm',
+    'low_quality': 'l'
+}
```
Comment (leotrs) [inline manim/constants.py:136]: Did you run black on this? In any case, please leave a blank line at the end of the file.

File: manim/config/config_utils.py [hunk 3]
```diff
@@ -349,25 +349,27 @@ def _parse_cli(arg_list, input=True):
     # The following are mutually exclusive and each overrides
     # FRAME_RATE, PIXEL_HEIGHT, and PIXEL_WIDTH,
     parser.add_argument(
-        "-l",
```
Comment (leotrs) [inline manim/config/config_utils.py:352]: I'm not sure we should completely remove these flags just yet. Doing so will most likely break every single user's workflow. Yes, it's an easy change, but it still needs to be done with care.

Is there a way of adding a deprecation warning instead?

**Comment by leotrs** [comment]
The test you added is looking good. However, this seems to have broken other existing tests. Could you please check those?

Finally, please make sure to rebuild the documentation as well. Specifically, this PR should change `https://manimce.readthedocs.io/en/latest/tutorials/configuration.html` and the doctest found therein.

**Comment by jvdoorn** [comment]
Yeah so, when I run just the new test it works. However, when running all tests it fails. I'm not sure what it is caused by, it seems that when running all tests that the parser expects different arguments. I was hoping one of you could enlighten me on the cause of the error it throws.

I'll update the docs!

**Comment by leotrs** [comment]
cc @ManimCommunity/tests 

Also, does anyone in @ManimCommunity/core object to this change? I think we should deprecate the current flags instead of removing them.

File: manim/config/config_utils.py (hunk 3 above)
Comment (jvdoorn) [reply manim/config/config_utils.py:352]: I don't think it is possible to tell `-l` apart from `--low_quality`. If we want to warn for deprecation we could maybe create a new argument with just `-l` and if someone uses it warn them. Not sure if that will produce clean code though.

Comment (leotrs) [reply manim/config/config_utils.py:352]: Hmm that's a good point. I'm still a bit weary of breaking every user's flow though. Let's see what others think.

**Comment by jvdoorn** [comment]
I have added a deprecation warning to the old flags but they are still supported. As far as I'm aware this PR is now complete and ready for review.

**Review by eulertour — COMMENTED** [review]
Thanks for writing tests!

**Comment by kilacoda** [comment]
CMIIW, but merging this means something like `-pm` will become `-pqm`, right? Or will they have to be separated like `-p -qm`?

**Comment by jvdoorn** [comment]
They don't have to be separated as long as -qm is the last argument. 

See [Python documentation](https://docs.python.org/3/library/argparse.html#option-value-syntax).

File: tests/test_scene_rendering/test_cli_flags.py [hunk 4]
```diff
@@ -65,3 +64,45 @@ def test_n_flag(tmp_path, simple_scenes_path):
     ]
     _, err, exit_code = capture(command)
     assert exit_code == 0, err
+
+
+def test_quality_flags():
```
Comment (huguesdevimeux) [inline tests/test_scene_rendering/test_cli_flags.py:69]: Tests are perfect but not at the right place; as your tests does not involve any video rendering.  

Move it to tests/test_cli_flags.py (or another name, idk). 

Maybe we should merge scene rendering tests and other tests, idk

Comment (jvdoorn) [reply tests/test_scene_rendering/test_cli_flags.py]: Done!

**Comment by jvdoorn** [comment]
Uhm, not sure why the tests fail but it seems unrelated to my changes?

**Comment by eulertour** [comment]
Probably transient errors. I tried re-running them, but if its still an unrelated failure we can merge anyway.

**Comment by jvdoorn** [comment]
Well they fail when installing, and since I changed nothing in the install process I'd say it is not possible that my changes broke the tests.

**Review by eulertour — APPROVED** [review]
Since the old flags are deprecated rather than removed I don't see why this can't be merged now.

**Review by kilacoda — APPROVED** [review]

**Review by huguesdevimeux — CHANGES_REQUESTED** [review]
Wait, there is not end to end test for these flags (except for -ql). 

Do you think it's the right PR for that ? I know, there are already tests for these flags but not invovling the whole video process. 
If you decide to do so, everything is explained in the wiki on github, [testing](https://github.com/ManimCommunity/manim/wiki/Testing) section

i mark it as change requested, just to have time to discuss a little bit

**Comment by eulertour** [comment]
Even if you do decide that you want to add end-to-end tests you shouldn't block this PR for it; that should be discussed separately.

**Comment by huguesdevimeux** [comment]
> Even if you do decide that you want to add end-to-end tests you shouldn't block this PR for it; that should be discussed separately.

That's why I was asking.

**Review by huguesdevimeux — APPROVED** [review]

**Pull request merged**